tests/gen_cert_test.py

```diff
@@ -181,3 +179,69 @@ def test_cert_date(self, force_date, locale, timezone, expected):
         """
         cert_date = get_cert_date(force_date, locale, timezone)
         assert_equal(expected, cert_date)
+
+
+class CertificateSubtemplateTests(unittest.TestCase):
+    """
+    Tests the subtemplates feature for certs
+
+    Subtemplates are currently written to key on designations only
+
+    These tests have a specific order
+    """
+    subtemplate_courses = [
+        'Test/Subtemplates/ByDesignations'
+    ]
+
+    def test_subtemplate_courses_exist(self):
+        """
+        Check that the subtemplate courses exist in settings.CERT_DATA
+        """
+        found_sub_course_ids = []
+        for course_id, course_data in settings.CERT_DATA.iteritems():
```

Comment:
whenever you test on CERT_DATA you should make a copy of it so it doesn't mess with that data for later tests:

```python
import copy
test_cert_data = copy.deepcopy(settings.CERT_DATA)
for course_id, course_data in test_cert_data.iteritems():
    ...
```